### PR TITLE
feat(list,kill): implement editor-instance discovery and termination

### PR DESCRIPTION
**Pull request by yukimemi**
## Summary

Implements the previously-stubbed `todoke list` and `todoke kill` subcommands.

### `todoke list`

Discovers running editor instances by reverse-engineering each `[todoke.<name>]`'s `listen` template:

1. Render the template once with a sentinel group → split on the sentinel to recover the (prefix, suffix) skeleton.
2. Walk the filesystem (Unix `read_dir`, Windows `FindFirstFileW` over `\.\pipe\*`) for candidates that fit the shape.
3. RPC-ping each candidate (500ms timeout) to mark `alive` vs `stale`.

Output is one row per instance: `status  target  group  listen-path`. `--alive-only` filters out stale candidates (e.g. socket files orphaned by a crashed nvim).

### `todoke kill`

- For **alive** instances: connect, send `:qall!`, sleep 800ms (`QUIT_GRACE`), re-ping. Returns `Quit` / `StillAlive` / `Forced { pid }`.
- For **stale** instances: `cleanup_stale` unlinks the socket file on Unix; on Windows it's a no-op (named pipes are handle-tied and can't outlive their owning process).
- `--force` escalates `StillAlive` to OS-level kill — `SIGKILL` on Unix (via `libc`), `TerminateProcess` on Windows (via `windows-sys`). PID is captured up-front via `vim.fn.getpid()` because the RPC connection drops as nvim exits.

```
$ todoke list
alive  nvim  default  /tmp/nvim-todoke-default.sock
stale  nvim  git      /tmp/nvim-todoke-git.sock

$ todoke kill --all --force
ok  nvim default /tmp/nvim-todoke-default.sock
ok  nvim git     /tmp/nvim-todoke-git.sock — stale, removed
```

### Design notes

- **Sentinel rendering** isolates the `{{ group }}` slot in the listen template without depending on its surface form. Templates with two `{{ group }}` substitutions, or none at all, are rejected with a clear error (the former because the group can't be recovered unambiguously, the latter because there's nothing to enumerate — a single-instance target).
- **`vars.*` references** in the listen template are stable across both sentinel renders, so they end up baked into the prefix/suffix as fixed text.
- **Windows named pipes** can't be staged with `File::create`, so the on-disk discovery integration test is `#[cfg(unix)]`. The Windows enumeration path is exercised via the existing `FindFirstFileW` plumbing during live-instance demos.
- **Force-kill PID retrieval** must happen *before* `:qall!`. After the command, the connection drops as nvim exits and the PID is unrecoverable.
- **`--force` is opt-in** because it discards autocmds, unsaved buffers, and any pre-quit cleanup logic.

### Testing

- 10 new unit tests in `registry::tests` (ListenSkeleton edge cases, discovery against tempdir entries, stale unlink).
- 5 new integration tests in `tests/cli.rs` (empty-config paths, Unix-only stale-socket cleanup loop end-to-end).
- `cargo make check` green: fmt + clippy + 75 unit + 21 integration + locked build.

## Test plan

- [x] `cargo make check` (pre-push hook) — green.
- [x] Live demo on Windows: spawn 2 headless nvim → `list` shows both alive → `kill <group>` removes one → `kill --all --force` removes the other → `list` shows none.
- [ ] Linux/macOS CI exercises the `#[cfg(unix)]` paths (stale socket cleanup, `read_dir` enumeration).
- [ ] Manual verification on Linux that `--force` actually escalates when `:qall!` is wedged (e.g. plugin with `VimLeavePre = function() while true do end end`).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Release Notes

* **New Features**
  * Added `--force` flag to the kill command for forceful termination when normal shutdown fails.
  * The list command now displays running editor instances with their current status.
  * The kill command is now fully functional with filtering and termination capabilities.

* **Tests**
  * Added CLI integration tests for list and kill functionality.

### CODE REVIEW

File: src/registry.rs [hunk 1]
```diff
@@ -0,0 +1,649 @@
...
+            out.push(Instance {
+                target: target_name.clone(),
+                group,
+                listen: path,
+                alive,
+            });
+        }
+    }
+    out.sort_by(|a, b| a.target.cmp(&b.target).then_with(|| a.group.cmp(&b.group)));
+    out
```
[the remark this answers is not in the document]
Comment (yukimemi) [reply src/registry.rs]: @gemini-code-assist Applied in 4b4e90d. `discover()` now uses `tokio::task::JoinSet` for the per-candidate ping fan-out — a tempdir full of stale sockets goes from `N × 500ms` sequential to a single 500ms parallel wave.

File: src/registry.rs [hunk 2]
```diff
@@ -0,0 +1,649 @@
...
+fn enumerate_candidates(skeleton: &ListenSkeleton) -> Vec<String> {
+    let prefix_path = Path::new(&skeleton.prefix);
+    let parent = prefix_path
+        .parent()
+        .filter(|p| !p.as_os_str().is_empty())
+        .unwrap_or(Path::new("."));
+    let basename_prefix = prefix_path
+        .file_name()
+        .and_then(|s| s.to_str())
+        .unwrap_or("");
```
[the remark this answers is not in the document]
Comment (yukimemi) [reply src/registry.rs]: @gemini-code-assist Applied in 4b4e90d. Detect the trailing-slash shape explicitly so `parent = prefix_path` and `basename_prefix = ""` for templates like `/tmp/{{ group }}.sock`. Dropped the `MAIN_SEPARATOR` fallback since this branch is `#[cfg(unix)]` and only `/` matters.

File: src/registry.rs [hunk 3]
```diff
@@ -0,0 +1,649 @@
...
+        None
+    };
+
+    let _ = nvim.command("qall!").await;
+    drop(nvim);
+
+    tokio::time::sleep(QUIT_GRACE).await;
+    if !ping(listen).await {
+        return Ok(KillOutcome::Quit);
+    }
```
[the remark this answers is not in the document]
Comment (yukimemi) [reply src/registry.rs]: @gemini-code-assist Applied in 4b4e90d. `kill_instance` now `timeout(QUIT_GRACE, io_handle)` after `qall!` instead of an unconditional sleep — clean quits return in milliseconds (when nvim drops the RPC connection), only a wedged process pays the full 800ms grace before we re-ping and decide whether to escalate.

File: src/registry.rs [hunk 4]
```diff
@@ -0,0 +1,649 @@
...
+    for ent in entries.flatten() {
+        let name = ent.file_name();
+        let Some(name_s) = name.to_str() else {
+            continue;
+        };
+        if name_s.starts_with(basename_prefix)
+            && name_s.ends_with(basename_suffix)
+            && name_s.len() > basename_prefix.len() + basename_suffix.len()
+        {
+            out.push(ent.path().to_string_lossy().into_owned());
```
[the remark this answers is not in the document]
Comment (yukimemi) [reply src/registry.rs:252]: @coderabbitai Fixed in 282f112. `enumerate_candidates` (Unix) now gates on `std::os::unix::fs::FileTypeExt::is_socket()` before pushing the candidate, so a regular file that happens to match the listen template's basename shape never surfaces and never gets unlinked by `cleanup_stale`. Added a focused unit test (`discover_filters_out_regular_files_matching_the_pattern`) that stages a real socket alongside a plain `.sock` file and verifies only the socket is discovered.

File: tests/cli.rs [hunk 5]
```diff
@@ -704,3 +704,152 @@ sync = false
...
+        "--todoke-config",
+        &config.to_string_lossy(),
+        "kill",
+        "--all",
+    ]);
+    assert!(ok, "stderr: {err}");
+    assert!(out.contains("stale, removed"), "stdout: {out}");
+    assert!(!stale_a.exists(), "default socket should be unlinked");
+    assert!(!stale_b.exists(), "git socket should be unlinked");
+}
```
[the remark this answers is not in the document]
Comment (yukimemi) [reply tests/cli.rs:887]: @coderabbitai Fixed in 282f112. Both integration tests now use a `make_stale_socket` helper that calls `UnixListener::bind(path)` and lets the listener drop in scope — the socket file remains on disk but no one accepts on it (canonical "stale socket" shape). Each test also keeps a regular-file decoy (e.g. `nvim-todoke-imposter.sock`) and asserts it does not surface, locking in the `is_socket()` gate from the sibling fix.

File: src/dispatcher.rs [hunk 6]
```diff
@@ -188,15 +188,119 @@ pub async fn doctor(cli: &Cli) -> Result<()> {
...
+        }
+        match crate::registry::kill_instance(&inst.listen, force).await {
+            Ok(crate::registry::KillOutcome::Quit) => {
+                println!("{}  {header}", styled("ok", level_ok()));
+            }
+            Ok(crate::registry::KillOutcome::Forced { pid }) => {
+                println!(
+                    "{}  {header} — forced (pid={pid})",
+                    styled("ok", level_ok()),
+                );
```
[the remark this answers is not in the document]
Comment (yukimemi) [reply src/dispatcher.rs:283]: @coderabbitai Fixed in 282f112. Folded `cleanup_stale(listen)` into `kill_instance` itself so the post-`Forced` cleanup is encapsulated with the kill — `dispatcher::kill` doesn't need to know that SIGKILL leaves residue. Errors from the cleanup are swallowed because the force-kill itself succeeded; any leftover stale entry will get picked up on the next `todoke kill` run.

File: src/registry.rs [hunk 7]
```diff
@@ -0,0 +1,649 @@
...
+    let prefix_path = Path::new(&skeleton.prefix);
+    let parent = prefix_path
+        .parent()
+        .filter(|p| !p.as_os_str().is_empty())
+        .unwrap_or(Path::new("."));
+    let basename_prefix = prefix_path
+        .file_name()
+        .and_then(|s| s.to_str())
+        .unwrap_or("");
+    let basename_suffix = skeleton.suffix.as_str();
```
[the remark this answers is not in the document]
Comment (yukimemi) [reply src/registry.rs:227]: @coderabbitai Fixed in 282f112 — to clarify, this concern was *not* fully addressed by 4b4e90d (that commit only handled the trailing-slash shape `/tmp/{{ group }}.sock`). The `/tmp/todoke/{{ group }}/nvim.sock` case you raised needs the suffix to forbid path separators outright, since enumeration walks one directory level only. `ListenSkeleton::from_template` now rejects any template whose suffix contains `/` with a clear error, surfacing the misconfiguration at parse time instead of producing zero hits silently. New unit test: `skeleton_errors_when_group_is_not_in_final_path_component`.
